core/src/main/java/org/infinispan/interceptors/impl/CacheLoaderInterceptor.java

```diff
@@ -483,12 +498,83 @@ public void disableStore(@Parameter(name = "storeType", description = "Fully qua
       persistenceManager.disableStore(storeType);
    }
 
-   private class WrappedEntrySet extends AbstractDelegatingEntryCacheSet<K, V> {
-      private final CacheSet<CacheEntry<K, V>> entrySet;
+   private ToIntFunction<Object> getSegmentMapper(Cache<?, ?> cache) {
+      DistributionManager dm = cache.getAdvancedCache().getDistributionManager();
+      if (dm != null) {
+         return dm.getCacheTopology()::getSegment;
```

Comment:
`CacheLoaderInterceptor` can inject the `KeyPartitioner` directly.

BTW, the `KeyPartitioner` is customizable, but we don't say in the javadoc whether what key format it's supposed to support. That might be because we didn't have any guarantee that it would only be called on the storage-formatted keys at the time we added it :)

Reply:
Thanks, forgot to change this one from before.